.github/workflows/build.yml

```diff
@@ -35,11 +35,15 @@ jobs:
           path: ~/.sonar/cache
           key: ${{ runner.os }}-sonar
           restore-keys: ${{ runner.os }}-sonar
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        if: ${{ env.SONAR_TOKEN != '' }}
```

Comment:
Please put the `if` expression on line `34` just before `with:`, that way it would be more readable.

Reply:
done